hal/common/file.hpp

```diff
@@ -65,7 +65,7 @@ class File
 	 * 
 	 * \return  success
 	 */
-	virtual bool open(const char* path) = 0;
+	virtual bool open(const char* path, bool new_file) = 0;
```

Comment:
Then I have to remove the call to the function open from the constructor of File_linux class.